cmake/CompilerFlags.cmake

```diff
@@ -64,6 +64,11 @@ if(MSVC AND NOT ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")) # Visual C++ (VS
   target_compile_options(project_options INTERFACE $<$<CONFIG:Debug>:/DMSVC_DEBUG>) # Triggers code in main.cc to catch floating point NaNs
 elseif(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang") # g++/Clang
 
+  # TODO: after we fix all test, enable this by default on Debug builds
+  # option(FORCE_DEBUG_ARITHM_GCC_OR_CLANG "Enable trapping floating point exceptions in non Debug mode" OFF)
+  option(FORCE_DEBUG_ARITHM_GCC_OR_CLANG "Enable trapping floating point exceptions" OFF)
+  mark_as_advanced(FORCE_DEBUG_ARITHM_GCC_OR_CLANG)
```

Comment:
Disable Floating point exceptions in Debug mode on Unix (clang/gcc) for now until all NaN in unit tests are fixed

Reply:
Yes, this will be great to re-enable.  I will try to just make a branch right away where I re-enable it and see the 19 remaining failures so I don't lose focus on it.